utils/analysis/tsh_traj_anal.py

```diff
@@ -0,0 +1,178 @@
+#!/bin/env python3
+# /// script
+# requires-python = ">=3.9"
+# # NOTE: PyQT6 dependency is a workaround for:
+# # https://github.com/astral-sh/uv/issues/6893
+# dependencies = [
+#     "matplotlib",
+#     "numpy",
+#     "PyQT6",
+# ]
+# ///
+"""Analysis script for Surface Hopping trajectories generated by ABIN
+
+Script for analyzing potential energy surfaces, energies and electronic populations along
+individual trajectories from ABIN's Fewest Switches or Landau-Zener surface hopping simulations.
+
+The script should be launched within the ABIN trajectory folder as:
+
+    python3 tsh_traj_analysis.py
+
+or:
+
+    ./tsh_traj_analysis.py
+
+
+You can also launch the script via 'uv' (https://github.com/astral-sh/uv)
+which will automatically install all needed dependencies (numpy, matplotlib)
+in a separate virtual environment:
+
+    uv run tsh_traj_analysis.py
+
+Author: Jiri Janos 2025
+"""
+
+import os.path
+
+# ABIN output files that we need
+POPULATION_FILE = "pop.dat"
+PES_FILE = "PES.dat"
+ENERGY_FILE = "energies.dat"
+
+AUtoEV = 27.2114
+
+
+def parse_cmd():
+    """Parse command-line arguments"""
+    import argparse
+
+    parser = argparse.ArgumentParser(
+        description="Analyze Surface Hopping trajectories",
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter,
+    )
+    parser.add_argument("-s", "--save-fig", action="store_true", help="Save png figure")
+    parser.add_argument("-eu", "--energy-units", choices=["eV", "au"], default="eV", help="Energy units")
+    parser.add_argument(
+        "-es",
+        "--energy-shift",
+        action="store_false",
+        help="Shift curves so that the lowest state minimum has 0 energy",
+    )
```

Comment:
This option is currently confusing to me. If I understand correctly, specifying this option actually **turns off** the energy shift?

What if we instead have an option that specifies the zero energy (probably in atomic units)? For example

```suggestion
    parser.add_argument(
        "-ze",
        "--zero-energy",
        type=float
        default=None,
        help="Specify zero energy (in atomic units). All energy curves will be shifted with respect to this energy.",
    )
```

To turn off energy shifting, the user then could pass (`--energy-shift 0`).
And this would also allow to pass any value, which would be handy if you were analyzing multiple trajectories and wanted to have consistent E0)

Reply:
That's a good point. I added it ages ago and probably used it once in my life. Your suggestion makes complete sense.